typhos/plugins/core.py

```diff
@@ -143,7 +145,24 @@ def cast(self, value):
                          dtype, self.signal.name, self.signal_type)
 
         logger.debug("Casting %r to %r", value, self.signal_type)
-        if self.signal_type is np.ndarray:
+        if self.enum_strs:
+            # signal_type is either int or str
+            # use enums to cast type
+            if self.signal_type is int:
+                # Get the index
+                try:
+                    value = self.enum_strs.index(value)
+                except (TypeError, ValueError, AttributeError):
+                    value = int(value)
+            elif self.signal_type is str:
+                # Get the enum string
+                try:
+                    value = self.enum_strs[value]
+                except (TypeError, ValueError):
+                    value = str(value)
+            else:
+                raise TypeError(f"Invalid combination: {self.enum_strs=} with {self.signal_type=}")
```

Comment:
TIL you could do this with f-strings.  This ends up also printing `self`, might not be the most helpful.  But this is truly a nitpick

Reply:
Oh shoot I didn't realize this would point self... I'll make some sort of adjustment

Reply:
Simple rework to put the variable names in myself:
```
TypeError: Invalid combination: enum_strs=('zero', 'one', 'two') with signal_type=<class 'float'>
```